# Aufgabenblatt 4: Arrays

## Aufgabe 1: Wiederholung Strings
### a) nutue print für die folgenden Beispiele:
- referenziere auf einen `exemplarischen Ordner-Pfad` z.B: c:\ordner\datei

In [89]:
print("c:\\ordner\\datei")

c:\ordner\datei


- erstelle einen String mit Sichtbaren Anführungszeichen

In [90]:
print("I\'m lazy")

I'm lazy


- erstelle einen String über mehrere Zeilen (inkl. Zeilenumbruch)

In [91]:
print("Zeilen\n Umbruch")

Zeilen
 Umbruch


### b) kombiniere String mit Variablen
- gebe einen String mit einer Variable in einer print Funktion gemeinsam aus. z.B. `I'm 35`

In [92]:
number = 35
print(f"I\'m {number}")

I'm 35


- gibt es eine andere Schreibweise

In [93]:
print(f"I\'m {35}")

I'm 35


### c) Verkettung und Bearbeitung von Strings
- erstelle einen String `string_1` mit dem Text `Python ist` mittels Anführungszeichen

In [94]:
string_1 = 'Python ist'
print(string_1)

Python ist


- erstelle einen String `string_2` mit dem Text `super` mittels einer Funkition

In [95]:
def get_super_string() -> str:
    return "super"

string_2 = get_super_string()
print(string_2)

super


- erstelle einmal einen String `string_3` mit dem Text `:)` nach deiner Wahl

In [96]:
string_3 = ":)"

- füge die drei Strings zu einem String `string_all` zusammen

In [97]:
print(string_1, string_2, string_3)

Python ist super :)


- schreibe mittels Multiplikator den folgenden `Python ist super super super :)` in die Variable `string_multi`

In [98]:
print(string_1, (string_2 + " ") * 3, string_3)

Python ist super super super  :)


## Aufgaben 2: Bearbeiten von Strings
### a) Grundfunktionen

Analysiere den String ``string_find`` und bearbeite die nachfolgenden Schritte 

In [99]:
# String für weitere Bearbeitung erstellt
string_find = "  THIS and that and borring stuff! bla "

- gebe die `Länge` des Strings `string_find` an
<br> Ergebnis 39

In [100]:
print(len(string_find))

39


- gebe die `erste Position` des Buchstabens `a` aus dem String `string_find` an
<br> Ergebnis 7

In [101]:
print(string_find.index("a"))

7


- prüfe, ob der String `string_find` mit `tha` beginnt

In [102]:
print(string_find.startswith("tha"))

False


- `veränadere` den Buchstaben `a` in `A` über alle Elemente in dem String `string_find`

In [103]:
print(string_find.replace("a", "A"))

  THIS And thAt And borring stuff! blA 


- prüfe, ob der String `string_find` mit den der String mit `ff` endet

In [104]:
print(string_find.endswith("ff"))

False


- gebe den String `string_find` nur mit `klein Buchstaben` aus (lower case)

In [105]:
print(string_find.lower())

  this and that and borring stuff! bla 


- Zähle wie viele `a` es in dem String `string_find` gibt?
<br> Ergebnis 4

In [106]:
print(string_find.count("a"))

4


- gebe den String `string_find` nur mit `gross Buchstaben` aus

In [107]:
print(string_find.upper())

  THIS AND THAT AND BORRING STUFF! BLA 


- gebe die Stelle des `letztens` `a` innerhalb des Strings `string_find` an
<br> Ergebnis 37

In [108]:
print(string_find.rfind("a"))

37


- `enterferne alle Leerzeichen` aus dem String `string_find`

In [109]:
print(string_find.replace(" ", ""))

THISandthatandborringstuff!bla


- `lösche **nur** alle Leerzeichen am Anfang und am Ende` aus dem String `string_find` und 
- Zähle alle `Elemente` des Strings und übergebe das Ergebnis in eine Variable `string_find_len`.
- gebe den neuen `string_find` und das Ergebnis mittels print aus
<br> Ergebnis 36

In [110]:
string_find_len = len(string_find.strip(" "))
print(string_find.strip(" "), string_find_len)

THIS and that and borring stuff! bla 36


- nun `lösche nur die ersten Leerzeichen am Anfang` aus dem String `string_find` und 
- übergebe das Ergebnis, vom zählen aller Elemente in einer neuen Variable wieder. Wie ist die `Länge`?
<br> Ergebnis 38

In [111]:
print(len(string_find.lstrip(" ")))
print(string_find.lstrip(" "))

# Ich glaube das Ergebnis 38 ist falsch -> Es sind 2 Leerzeichen am anfang

37
THIS and that and borring stuff! bla 


### b) String zusammenführen

- Erstelle die beiden Strings jeweils in einer Variabeln
    - `string_1` = "Python is"
    - `string_2` = "amazing!"
- führe anschliessend die beiden Strings in eine Variable zusammen, sodass `Python is amazing!` herauskommt
- Gibt es mehr als eine Möglichkeit?


In [112]:
string_1 = "Python is"
string_2 = "amazing!"
print(string_1, string_2)
print(string_1 + " " + string_2)
print(f"{string_1} {string_2}")

Python is amazing!
Python is amazing!
Python is amazing!


- verändere mittels einer `build in Funktion` in der Variable `string_2`. Es soll der Begriff `amazing` durch `booring`ausgetauscht werden
- gebe das Ergebnis mittels print aus

In [113]:
string_2 = string_2.replace("amazing", "boring")
print(string_1, string_2)

Python is boring!


#### c) Grundaufgaben Erweiterung

`string_find` (aus a) soll wie folgt bearbeitet werden. Die Bearbeitungen sollen in der Variable `liste_string_find` gespeichert werden:
- alle Leerzeichen am Ende und am Anfang aus `string_find` sollen enttfernt werden
- anschliessend sollen alle Buchstaben des Strings `string_find` **klein** geschrieben werden
- erstelle Listenelemente: d.h. jedes einzelne Wort aus dem String `string_find` soll ein eigenes Element in einer Liste werden
- stelle sicher, dass die Wörter in der Liste einmalig sind - d.h. ein Wort kann nicht mehrmals vor kommen
- Füge nun der Liste `liste_string_find` das Wort `is` hinzu. 
- Gebe die Liste mittles der print Funktion aus.
<br><br> **Ergebnis**: 'that', 'this', 'bla', 'borring', 'stuff!', 'and', 'is' -> Rheinfolge kann abweichen!!!

In [114]:
liste_string_find = string_find.strip(" ").lower().split(" ")
print(liste_string_find)
liste_string_find = list(dict.fromkeys(liste_string_find))
print(liste_string_find)
liste_string_find.append("is")
print(liste_string_find)


['this', 'and', 'that', 'and', 'borring', 'stuff!', 'bla']
['this', 'and', 'that', 'borring', 'stuff!', 'bla']
['this', 'and', 'that', 'borring', 'stuff!', 'bla', 'is']


- erstelle aus der Liste mittels Slicen den Satz: `this is borring` und gebe diesen aus

In [115]:
print(liste_string_find[0], liste_string_find[6], liste_string_find[3])

this is borring


- erstelle aus der Liste den Satz: `blablablablablabla` --> bla kommt 6 mal vor

In [116]:
print(liste_string_find[5] * 6)

blablablablablabla


- wie häufig ist `bla` in `blablablablablabla` enthalten --> lass den code zählen

In [117]:
print((liste_string_find[5] * 6).count("bla"))

6


### d) Was sind die Unterschiede bei split?
- Was ist der unterschied zwischen `split(sep = None)`, `split()` and `split(sep = " ")`?
<br> <br> Unteruche es an dem String `split_string`

In [118]:
split_string = "String mit  unterschiedlichen   Anzahl an  Leerzeichen."

In [119]:
print(split_string.split(sep=None))
print(split_string.split())
print(split_string.split(sep=" "))

['String', 'mit', 'unterschiedlichen', 'Anzahl', 'an', 'Leerzeichen.']
['String', 'mit', 'unterschiedlichen', 'Anzahl', 'an', 'Leerzeichen.']
['String', 'mit', '', 'unterschiedlichen', '', '', 'Anzahl', 'an', '', 'Leerzeichen.']


### e) Build-In Function "format" 
#### bearbeiten von Strings mittels format. Diese Funktion ist z.B. sehr hilfreich im Zusammenhang bei `print`
Befülle den Text `"Mein Name ist ... und ich bin ... Jahre alt. Die Aufgaben ist ..."`
- einmal mit den Variablen name, age, value für die Punkte

In [120]:
name = "Kai"
age = 35
value = "leicht"

print("Mein Name ist {} und ich bin {} Jahre alt. Die Aufgaben ist {}.".format(name, age, value))

Mein Name ist Kai und ich bin 35 Jahre alt. Die Aufgaben ist leicht.


- neben den Variabeln aus der Aufgabe davor, kann man format auch mit Positionsbestummungen (index) nutzen. Schreibe nun den Code von der AUfgabe davor um für die Indexbestimmung

In [121]:
print("Mein Name ist {0} und ich bin {1} Jahre alt. Die Aufgaben ist {2}.".format(name, age, value))

Mein Name ist Kai und ich bin 35 Jahre alt. Die Aufgaben ist leicht.


- schriebe den Code von oben nun ohne Variabeln oder Index. **Tipp** Variabeln bezeichnungen sind direkt im der format Funktion 

In [122]:
print("Mein Name ist {name} und ich bin {age} Jahre alt. Die Aufgaben ist {value}.".format(name = "Kai", age = 35, value = "leicht"))

Mein Name ist Kai und ich bin 35 Jahre alt. Die Aufgaben ist leicht.


 `Inhaltsfragen`
- Was passiert wenn die Reihnfolge im format jeweils für die drei Codearten oben geändert werden?
- Rheinfolge ist jetzt, `leicht, Kai, 35`

In [123]:
print("Mein Name ist {} und ich bin {} Jahre alt. Die Aufgaben ist {}.".format(value, name, age))
print("Mein Name ist {0} und ich bin {1} Jahre alt. Die Aufgaben ist {2}.".format(value, name, age))
print("Mein Name ist {name} und ich bin {age} Jahre alt. Die Aufgaben ist {value}.".format(value = "leicht", name = "Kai", age = 35))

Mein Name ist leicht und ich bin Kai Jahre alt. Die Aufgaben ist 35.
Mein Name ist leicht und ich bin Kai Jahre alt. Die Aufgaben ist 35.
Mein Name ist Kai und ich bin 35 Jahre alt. Die Aufgaben ist leicht.


`Index` geht auch richtig - Wie?

In [124]:
print("Mein Name ist {1} und ich bin {2} Jahre alt. Die Aufgaben ist {0}.".format(value, name, age))

Mein Name ist Kai und ich bin 35 Jahre alt. Die Aufgaben ist leicht.


 `Inhaltsfragen`
 - Was passiert wenn nicht alle Elemente übergeben werden?

In [125]:
#print("Mein Name ist {1} und ich bin {2} Jahre alt. Die Aufgaben ist {0}.".format(value, name))

### e) quote 007
- schreibe den Satz: `"My name is James, James Bond"` verwende hierfür **format** und gebe nur einen Parameter den Namen `James` wieder

In [126]:
print("My Name is {name}, {name} Bond".format(name="James"))
quote007 = "My Name is {name}, {name} Bond".format(name="James")

My Name is James, James Bond


### f) Aufgabe use: zip
- kombieniere die beiden nachfolgenden Listen mittels eines for-loops
- das Ergebnis soll u.a. sein: `Kai ist 36 Jahre alt`
<br><br> names_liste = ['Kai', 'James', 'Hulk']
<br> age_liste = [35, 42, '38']

In [127]:
names_liste = ["Kai", "James", "Hulk"]
age_liste = [35, 42, 38]
for i in range(3):
    print(f"{names_liste[i]} is {age_liste[i]} years old.")

Kai is 35 years old.
James is 42 years old.
Hulk is 38 years old.


### g) Strings justieren anhand der Länge
- Gebe die einzelnen Elemente der Liste `string_liste_just` aus? 

In [128]:
string_liste_just = ["kurz", "mittel lang", "der längste String"]

In [129]:
for e in string_liste_just:
    print(e)

kurz
mittel lang
der längste String


- Nun sollen die Elemente der `string_liste_just`  **`rechtsbündig`** ausgegeben werden (an dem maximalen Wert)
<br><br>
**Tipps:**
- kreiere eine Variable max_len
- du musst die Ausrichtung mitgeben werden --> Bestimme die maximale Länge der Elemente im Array --> mittels for-loop
- in einem zweiten foor-loop kannst du die justierung vornehmen mittels rjust

In [130]:
max_len = 0
for e in string_liste_just:
    if len(e) > max_len:
        max_len = len(e)

for e in string_liste_just:
    print(e.rjust(max_len))

              kurz
       mittel lang
der längste String


### h) Integer justieren auf Punkt Ebene

In [131]:
punkt_daten = [3.21, 30.4, 2.575, 100.1001]
punkt_daten

[3.21, 30.4, 2.575, 100.1001]

- Gebe die einzelnen Elemente der Liste `string_liste_just` aus? 

In [132]:
for e in punkt_daten:
    print(e)

3.21
30.4
2.575
100.1001


- gebe die Liste `punkt_daten` so aus, dass die Punkte auf der `gleichen höhe` sind
<br><br>
**Tipp**
- ähnlich zu der Aufgabe davor
- es müssen aber die Indexes oder länge von ELementen vor und nach dem Punkt bestimmt werden 

In [133]:
max_len_before = 0
for e in punkt_daten:
    if len(str(e).split(".")[0]) > max_len_before:
        max_len_before = len(str(e).split(".")[0])


for e in punkt_daten:
    before, after = str(e).split(".")
    print(before.rjust(max_len_before) + "." + after)


  3.21
 30.4
  2.575
100.1001


### i) Aufgabe: Operatoren 
- benutze Operatoren um zu prüfen, ob der Buchstabe `z` in dem Text aus der `Aufgabe e (Quote 007)` enthalten ist 

In [134]:
"z" in quote007

False

- ist der Buchstabe `d` in dem Text aus der `Aufgabe e (Quotw 007)` enthalten?

In [135]:
"d" in quote007

True

## Aufgabe 3: Listen bearbeiten
### a) Benutze verschiedene Funktionen um eine Liste zu bearbeiten

- erstelle zuerst eine leere Liste mit dem Namen `list_example`

In [136]:
list_example = []
print(list_example)

[]


- füge der Liste den Wert `1` hinzu

In [137]:
list_example.append(1)
print(list_example)

[1]


- füge nun der Liste den Wert `3, 5, 7, 2, 13, 17` am Listenende hinzu --> also nach `1`

In [138]:
for e in [3,5,7,2,13,17]:
    list_example.append(e)
print(list_example)

[1, 3, 5, 7, 2, 13, 17]


- füge der Liste den Wert `2` nach dem Wert `5` (bzw. an dem Index 2) hin zu

In [139]:
list_example.insert(3, 2)
print(list_example)

[1, 3, 5, 2, 7, 2, 13, 17]


- füge der Liste die folgenden Integerwerte `39, 27, 21` am Listenende hinzu

In [140]:
for e in [39,27,21]:
    list_example.append(e)
print(list_example)

[1, 3, 5, 2, 7, 2, 13, 17, 39, 27, 21]


- entferne nur das `letzte` Element aus der Liste

In [141]:
list_example.pop()
print(list_example)

[1, 3, 5, 2, 7, 2, 13, 17, 39, 27]


- entfenre das Element `7` aus der Liste

In [142]:
list_example.remove(7)
print(list_example)

[1, 3, 5, 2, 2, 13, 17, 39, 27]


- entferne das `zweite Element` aus der Liste

In [143]:
list_example.pop(1)
print(list_example)

[1, 5, 2, 2, 13, 17, 39, 27]


- sortiere die Liste (beginnend mit dem geringsten Wert)

In [144]:
list_example.sort()
print(list_example)

[1, 2, 2, 5, 13, 17, 27, 39]


### b) Bilde die Potenz der Listenwerte 

- Bilde für jedes Element der Liste `basis_liste` die dazugehörige Potenz
- schriebe den neuen Wert in eine neue Liste
- Sortiere das Ergebnis (beginnend mit dem höchsten Wert)

In [145]:
basis_liste = [3, 2, 9, 5, 7, 4, 1]
basis_liste

[3, 2, 9, 5, 7, 4, 1]

In [146]:
pot_list = []
for e in basis_liste:
    pot_list.append(e**e)
pot_list.sort(reverse=True)
print(pot_list)

[387420489, 823543, 3125, 256, 27, 4, 1]


### c) Erstelle Spielpaarungen

- Es gibt die Mannschaften `Norwegen, Schweden, Finnland, Dänemark`
- Diese Mannschaften sollen jeweils gegen einander spielen.
- erstelle alle verschiedenen möglichen Spielpaarungen (**Tipp: enumerate**) und gebe diese anschliessend aus

In [147]:
teams = ["Norwegen", "Schweden", "Finnland", "Dänemark"]

for i, team1 in enumerate(teams):
    for team2 in teams[i + 1:]:
        print(team1, "vs", team2)


Norwegen vs Schweden
Norwegen vs Finnland
Norwegen vs Dänemark
Schweden vs Finnland
Schweden vs Dänemark
Finnland vs Dänemark


- gebe automatisiert mittels for Loop und den Variabeln `team1 vs team2` die zuvor erstellten Spielpaarungen aus

### d) Listen anpassen

- Die `liste_a` ist die Basis Liste
- Wenn der Wert von `liste_a` in Liste `liste_b` enthalten ist, soll der Wert aus `liste_b` direkt nach dem gleichen Element in `listen_a` angefügt werden
- Wenn nicht soll der Wert an das Ende der `liste_a` hinzugefügt werden.
<br><br>
**Tipp**
- Verwende ein for-loop und eine if-else-cause
<br><br> Die Lösung sieht wie folgt aus: *['a', 'a', 'a', 'c', 'c', 'e', 'e', 'g', 'g', 'f', 'f', 'd', 'd', 'h']*

In [148]:
liste_a = ['a','c', 'e', 'g']
liste_b = ['c', 'f', 'd', 'h', 'a', 'e', 'd', 'a', 'f', 'g']

In [149]:
for e in liste_b:
    if e in liste_a:
        index = liste_a.index(e)
        liste_a.insert(index + 1, e)

print(liste_a)

['a', 'a', 'a', 'c', 'c', 'e', 'e', 'g', 'g']


### e) Elementensuche bei einer bestimmten Bedingung in einer Liste

- Gebe alle Elemente der Liste `search_list` aus, welche das Teilwort / Buchstabenfolge `und` in ihrem Wort enthalten haben
- schreibe nur die Wörter, welche ein `und` enthalten haben in eine neue Liste

In [150]:
search_list = ['Mund', 'Lörrach', 'Hund', 'DHBW', 'rund', 'bunt', 'und']
search_list

['Mund', 'Lörrach', 'Hund', 'DHBW', 'rund', 'bunt', 'und']

In [151]:
new_list = []
for e in search_list:
    if "und" in e:
        new_list.append(e)

print(new_list)


['Mund', 'Hund', 'rund', 'und']


## Aufgabe 4: Dictionaries bearbeiten
### a) Nutze Funktionen um eine Dictionarie zu bearbeiten

- definiere ein Dictionarie mit dem Namen `work_dict` mit den `Schlüssel:Wert-Paaren` 
<br> **name** = Kai as *str* <br> **age** = 35 as *int* <br> **3** = 2.5 as *float*

In [152]:
work_dict = {"Name": "Kai", "Age": 35, "3":2.5}

- Gebe den Wert zu dem Schlüssel `age` aus

In [153]:
print(work_dict.get("Age"))

35


- füge das Element `uni=dhbw` dem Dictionarie hinzu

In [154]:
work_dict["Uni"] = "DHBW"
print(work_dict)

{'Name': 'Kai', 'Age': 35, '3': 2.5, 'Uni': 'DHBW'}


- verändere die folgende Wert und bestimme die dazugehörigen Schlüssel:
    - `3` zu `1,6` **und**  
    - `uni` zu `DHBW Lörrrach` im dictionarie

In [155]:
work_dict["3"] = 1.6
work_dict["Uni"] = "DHBW Lörrach"
print(work_dict)

{'Name': 'Kai', 'Age': 35, '3': 1.6, 'Uni': 'DHBW Lörrach'}


- entferne den Schlüssel `age` aus dem Dictionarie

In [156]:
work_dict.pop("Age")
print(work_dict)

{'Name': 'Kai', '3': 1.6, 'Uni': 'DHBW Lörrach'}


- gebe alle Schlüssel aus dem Dictionarie wieder

In [157]:
print(work_dict.keys())

dict_keys(['Name', '3', 'Uni'])


- gebe alle Werte aus dem Dictionarie wieder 

In [158]:
print(work_dict.values())

dict_values(['Kai', 1.6, 'DHBW Lörrach'])


- gebe alle Schlüssel-Wert-Paar wieder

In [159]:
print(work_dict)

{'Name': 'Kai', '3': 1.6, 'Uni': 'DHBW Lörrach'}


- itteriere nun über alle Schlüsselelemente des Dictionaries und gebe die jeweiligen dazugehörigen Werte aus

In [160]:
for key, value in work_dict.items():
    print(f"Key: {key}, Value: {value}")

Key: Name, Value: Kai
Key: 3, Value: 1.6
Key: Uni, Value: DHBW Lörrach


### b) Lohnverteilung für Postionen

- schreibe ein Dictionarie, indem der CEO = 10.000, der Tramleiter = 5.000, der Mitarbeiter = 2.5000 und der Student = 1250 verdient

In [161]:
salary_dict = {"CEO": 10000, "Tramleiter": 5000, "Mitarbeiter": 2500, "Student": 1250}
print(salary_dict)

{'CEO': 10000, 'Tramleiter': 5000, 'Mitarbeiter': 2500, 'Student': 1250}


- bestimme für die Position `Mitarbeiter` das Gehalt und gebe es als String folgendes mittels format aus: `Position` verdient `Summe` Wobei sowohl Position als auch Summe eine Variable sind

In [162]:
position = "Mitarbeiter"
salary = salary_dict[position]
print(f"{position} verdient: {salary}")

Mitarbeiter verdient: 2500


- Aufgrund der derzeitig hohen Inflation sollen alle Gehälter um 9 % steigen. 
- Berechne die neuen Gehälter im Dictionarie und speicher den neuen Wert im Dictionarie
<br> `verwende eine for Schleife hier und nicht update`

In [163]:
for key, value in salary_dict.items():
    salary_dict[key] *= 1.09

print(salary_dict)


{'CEO': 10900.0, 'Tramleiter': 5450.0, 'Mitarbeiter': 2725.0, 'Student': 1362.5}


## Aufgabe 5: Bearbeiten von Mengen
### a) Funktionen um eine Set zu bearbeiten
- definiere ein **Set** `set_one` mit den Werten `1, 2, 3, 1, 4, 3, 2, 8`

In [164]:
set_one = {1,2,3,1,4,3,2,8}
print(set_one)

{1, 2, 3, 4, 8}


- definiere nun ein weiteres **Set** `set_two` mit den Werten `1, 1, 6, 6, 4, 3, 5, 7`

In [165]:
set_two = {1,1,6,6,4,3,5,7}
print(set_two)

{1, 3, 4, 5, 6, 7}


- füge nun `set_one` den Wert `5` hinzu

In [166]:
set_one.add(5)
print(set_one)

{1, 2, 3, 4, 5, 8}


- entferne anschliessend aus dem `set_one` den Wert `2`

In [167]:
set_one.remove(2)
print(set_one)

{1, 3, 4, 5, 8}


- entferne aus `set_one` den Wert `8` mit einer **anderen** Funktion

In [168]:
set_one.discard(8)
print(set_one)

{1, 3, 4, 5}


- gebe nun die beiden sets einmal aus

In [169]:
print(set_one)
print(set_two)

{1, 3, 4, 5}
{1, 3, 4, 5, 6, 7}


- verbinde die beiden Sets `set_one und set_two` zu einem gemeinsamen Set `set_union`

In [170]:
set_union = set.union(set_one, set_two)
print(set_union)

{1, 3, 4, 5, 6, 7}


- Bestimme nun die Gemeinsamenwerte aus `set_one und set_two`

In [171]:
print(set.intersection(set_one, set_two))

{1, 3, 4, 5}


- gebe nun die Werte aus, welche nicht in beiden sets vorkommen
- Die Basis für die nicht Gemeinsamkeiten soll `set_two` sein

In [172]:
print(set.difference(set_two, set_one))

{6, 7}


- Überprüfe, ob es eine leere Schnittmenge gibt

In [173]:
print(set.isdisjoint(set_one, set_two)) #True -> Schnittmenge ist leer | False -> es gibt mindestens ein gemeinsames element

False


- Überprüfe, ob `set_one` eine Untermenge von `set_two` ist

In [174]:
print(set.issubset(set_one, set_two))

True


- Überprüfe, ob `set_one` eine Obermenge von `set_two` ist

In [175]:
print(set.issuperset(set_one, set_two))

False
